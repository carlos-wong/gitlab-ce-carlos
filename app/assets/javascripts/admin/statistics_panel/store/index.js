import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';

Vue.use(Vuex);

export default () =>
  new Vuex.Store({
    actions,
    getters,
    mutations,
    state: state(),
  });
