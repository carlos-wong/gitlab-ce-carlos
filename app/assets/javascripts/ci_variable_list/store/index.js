import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import mutations from './mutations';
import state from './state';

Vue.use(Vuex);

export default (initialState = {}) =>
  new Vuex.Store({
    actions,
    mutations,
    state: {
      ...state(),
      ...initialState,
    },
  });
