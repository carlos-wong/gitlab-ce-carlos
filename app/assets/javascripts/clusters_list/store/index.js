import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import mutations from './mutations';
import * as actions from './actions';

Vue.use(Vuex);

export const createStore = initialState =>
  new Vuex.Store({
    actions,
    mutations,
    state: state(initialState),
  });

export default createStore;
