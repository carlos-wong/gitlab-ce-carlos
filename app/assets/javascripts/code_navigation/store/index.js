import Vuex from 'vuex';
import createState from './state';
import actions from './actions';
import mutations from './mutations';

export default new Vuex.Store({
  actions,
  mutations,
  state: createState(),
});
