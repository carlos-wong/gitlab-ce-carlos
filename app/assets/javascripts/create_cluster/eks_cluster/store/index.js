import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';

import clusterDropdownStore from './cluster_dropdown';

import {
  fetchRoles,
  fetchRegions,
  fetchKeyPairs,
  fetchVpcs,
  fetchSubnets,
  fetchSecurityGroups,
} from '../services/aws_services_facade';

const createStore = ({ initialState }) =>
  new Vuex.Store({
    actions,
    getters,
    mutations,
    state: Object.assign(state(), initialState),
    modules: {
      roles: {
        namespaced: true,
        ...clusterDropdownStore({ fetchFn: fetchRoles }),
      },
      regions: {
        namespaced: true,
        ...clusterDropdownStore({ fetchFn: fetchRegions }),
      },
      keyPairs: {
        namespaced: true,
        ...clusterDropdownStore({ fetchFn: fetchKeyPairs }),
      },
      vpcs: {
        namespaced: true,
        ...clusterDropdownStore({ fetchFn: fetchVpcs }),
      },
      subnets: {
        namespaced: true,
        ...clusterDropdownStore({ fetchFn: fetchSubnets }),
      },
      securityGroups: {
        namespaced: true,
        ...clusterDropdownStore({ fetchFn: fetchSecurityGroups }),
      },
      instanceTypes: {
        namespaced: true,
        ...clusterDropdownStore({ initialState: { items: initialState.instanceTypes } }),
      },
    },
  });

export default createStore;
