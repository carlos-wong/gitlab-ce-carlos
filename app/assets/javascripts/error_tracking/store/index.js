import Vue from 'vue';
import Vuex from 'vuex';

import * as actions from './actions';
import mutations from './mutations';

import * as listActions from './list/actions';
import listMutations from './list/mutations';
import listState from './list/state';

import * as detailsActions from './details/actions';
import detailsMutations from './details/mutations';
import detailsState from './details/state';
import * as detailsGetters from './details/getters';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      list: {
        namespaced: true,
        state: listState(),
        actions: { ...actions, ...listActions },
        mutations: { ...mutations, ...listMutations },
      },
      details: {
        namespaced: true,
        state: detailsState(),
        actions: { ...actions, ...detailsActions },
        mutations: { ...mutations, ...detailsMutations },
        getters: detailsGetters,
      },
    },
  });

export default createStore();
