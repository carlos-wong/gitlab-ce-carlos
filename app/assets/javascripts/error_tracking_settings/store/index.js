import Vue from 'vue';
import Vuex from 'vuex';
import createState from './state';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';

Vue.use(Vuex);

export default () =>
  new Vuex.Store({
    state: createState(),
    actions,
    getters,
    mutations,
  });
