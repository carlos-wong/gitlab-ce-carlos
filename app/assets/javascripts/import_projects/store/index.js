import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';

Vue.use(Vuex);

export { state, actions, getters, mutations };

export default () =>
  new Vuex.Store({
    state: state(),
    actions,
    mutations,
    getters,
  });
