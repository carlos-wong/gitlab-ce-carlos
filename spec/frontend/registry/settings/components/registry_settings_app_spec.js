import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import component from '~/registry/settings/components/registry_settings_app.vue';
import { createStore } from '~/registry/settings/store/';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Registry Settings App', () => {
  let wrapper;
  let store;
  let fetchSpy;

  const findSettingsComponent = () => wrapper.find({ ref: 'settings-form' });
  const findLoadingComponent = () => wrapper.find({ ref: 'loading-icon' });

  const mountComponent = (options = {}) => {
    fetchSpy = jest.fn();
    wrapper = shallowMount(component, {
      store,
      methods: {
        fetchSettings: fetchSpy,
      },
      ...options,
    });
  };

  beforeEach(() => {
    store = createStore();
    mountComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('call the store function to load the data on mount', () => {
    expect(fetchSpy).toHaveBeenCalled();
  });

  it('renders a loader if isLoading is true', () => {
    store.dispatch('toggleLoading');
    return wrapper.vm.$nextTick().then(() => {
      expect(findLoadingComponent().exists()).toBe(true);
      expect(findSettingsComponent().exists()).toBe(false);
    });
  });
  it('renders the setting form', () => {
    expect(findSettingsComponent().exists()).toBe(true);
  });
});
