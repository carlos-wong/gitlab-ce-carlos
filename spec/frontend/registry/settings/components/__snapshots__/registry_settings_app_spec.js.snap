// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry Settings App renders 1`] = `
<div>
  <p>
    
    Tag expiration policy is designed to:
  
  </p>
   
  <ul>
    <li>
      Keep and protect the images that matter most.
    </li>
     
    <li>
      
      Automatically remove extra images that aren't designed to be kept.
    
    </li>
  </ul>
   
  <settings-form-stub />
</div>
`;
